src/sagas.js

```diff
@@ -0,0 +1,24 @@
+import {all, call, put, takeEvery} from 'redux-saga/effects'
+import createSagaMiddleware from 'redux-saga';
+import {rootReducer} from 'reducers/rootReducer';
+import {applyMiddleware, createStore} from 'redux';
+import {watchSpinner} from 'components/Spinner/SpinnerSagas';
+import {signIn} from 'components/App/AppSagas';
+
+const sagaMiddleware = createSagaMiddleware();
+
+export const store = createStore(
+  rootReducer,
+  applyMiddleware(sagaMiddleware)
+);
+
+sagaMiddleware.run(rootSaga);
+
+export const action = (type, data = null) => store.dispatch({type, data});
+
+export function* rootSaga() {
+  yield all([
+    signIn(),
```

Comment:
Not sure `sagas.js` is the right place for calling `signIn()`.

Except this line, all the surrounding code is a boilerplate config for the saga library itself.

But `signIn()` is more about starting the application. The first action that is dispatched.

Or is it obvious that `rootSaga()` in `sagas.js` is the place where app starts execution? Sorry didn't read the docs yet :)

Reply:
That is the place where our sagas start working. We have no user action to start check is user signed in, so we start it here.